File: 0035-search-insert-position/0035-search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int size = nums.size();
        int val = 0;
        int lo = 0; 
        int mid = 0;
        int hi = size-1;

        while (lo <= hi) {
            mid = (hi+lo) / 2;
            if (nums[mid] == target) {
                return mid;
            } else if (target > nums[mid]) {
                lo = mid + 1;
            } else {
                hi = mid - 1;
            }
        }

        return lo;
    }
};

File: 0035-search-insert-position/0035-search-insert-position.go
func searchInsert(nums []int, target int) int {
    numsSize := len(nums)
    var mid int = 0
    lo := 0
    hi := numsSize -1
    val := 0

    for lo <= hi {
        mid = (hi+lo) / 2
        val = nums[mid]
        if val == target {
            return mid
        } else if target > val {
            lo = mid + 1
        } else {
            hi = mid -1
        }

    }   

    return lo

}